### ISSUE TITLE
Saving existing object produces another document

### ISSUE THREAD

**Issue by jamesdoran** [sebastienros/yessql#32]
Applies to: 2.0.0-beta-0005

Loving this project so far!

I may be using the API incorrectly, but I'm getting some unexpected behaviour.

If I create a new object and save it to my SQLite db:

```
using (var session = _createSessionFactory())
    session.Save(o);
```

I get a document added to the document/content tables.

If I then retrieve that object:

```
using (var session = _createSessionFactory()) 
    o = await session.QueryAsync<MyObjectType>().FirstOrDefault();
```

Make some changes:

```
o.Property1 = "NewValue"
```

Then save it back:

```
using (var session = _createSessionFactory())
    session.Save(o);
```

There are then 2 documents in the document table. If I get a count of MyObjectType, I get 2.

Am I doing something wrong? Happy to update the wiki with an example if you can provide one please.

**Comment by sebastienros** [comment]
Reuse the same session you loaded the object with. If you Save the object on a different session, it will thing it's a new one. A session only tracks the objects it itself loaded.

**Comment by jamesdoran** [comment]
Thanks Sebastian. That makes a lot of sense. 

Sorry for the newbie question! 




On Wed, Oct 26, 2016 at 9:11 AM +1030, "Sébastien Ros" <notifications@github.com> wrote:












Closed #32.

**Comment by sebastienros** [comment]
I actually implemented it, the fix came for free while I was trying to fix another issue.
